dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DefaultUploadSchedulerStrategy.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.upload
+
+import com.datadog.android.core.configuration.UploadSchedulerStrategy
+import com.datadog.android.core.internal.configuration.DataUploadConfiguration
+import com.datadog.android.core.internal.data.upload.DataOkHttpUploader.Companion.HTTP_ACCEPTED
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToLong
+
+internal class DefaultUploadSchedulerStrategy(
+    internal val uploadConfiguration: DataUploadConfiguration
+) : UploadSchedulerStrategy {
+
+    private var currentDelayIntervalMs = uploadConfiguration.defaultDelayMs
+
+    // region UploadSchedulerStrategy
+
+    override fun getMsDelayUntilNextUpload(
+        uploadAttempts: Int,
+        lastStatusCode: Int?,
+        throwable: Throwable?
+    ): Long {
+        if (uploadAttempts > 0 && throwable == null && lastStatusCode == HTTP_ACCEPTED) {
+            decreaseInterval()
```

Comment:
Question, if the upload has succeeded after some failed attempts, why not just recover to the default delay immediately?

Reply:
So far I've kept the historical behaviour of our backoff strategy. We will have future tickets to make a different implementations